### PR TITLE
Refactor: Delegate all handling to core

### LINKED ISSUE #4: Support persisted queries
Verify if there is any problem when using persisted queries.

Add an example in the examples directory to demonstrate it works as expected. See https://github.com/mercurius-js/mercurius/blob/master/docs/persisted-queries.md

### PR DESCRIPTION
**Pull request by barelyhuman**
## Changes 

- Modifies the implementation to let core handle the separation (depends on https://github.com/barelyhuman/mercurius/pull/1 ]

## Test Modifications 
- All tests now use `method` and body type as supported by the `core` instead of re-implementing their own details 
- Modify tests to now have proper errors instead of coded anomalies

Upstream Dependency: https://github.com/mercurius-js/mercurius/pull/1080
Closes #4

### CODE REVIEW

**Comment by simoneb** [comment]
I like this approach more than the previous attempts 👌

**Comment by simoneb** [comment]
Remind me again why we needed POST instead of GET and why now GET is fine?

**Comment by barelyhuman** [comment]
> Remind me again why we needed POST instead of GET and why now GET is fine?

I don't know why we needed POST, i thought there was a reason for it. I knew why the core was using `POST` thought. 

Now it's fine since I found out that there were no specific reasons for the decision to use POST. 
Since, the core registers both GET and POST requests and will handle persisted queries on both GET and POST, it's okay to do this.

**Comment by simoneb** [comment]
@barelyhuman after updating the latest version of mercurius, which includes the change we needed, CI is failing. Not sure if you have a chance to look into this, but there must be something that was overlooked here.

**Comment by barelyhuman** [comment]
It seems to be reusing the existing mercurius instance, a teardown should help. I'll check it out in a bit

**Comment by simoneb** [comment]
> It seems to be reusing the existing mercurius instance, a teardown should help. I'll check it out in a bit

Thanks 🙏

File: index.js [hunk 1]
```diff
@@ -39,35 +38,13 @@ async function mercuriusDynamicSchema(fastify, opts) {
...
-
-            return reply.graphql(
-              query,
-              req[kRequestContext],
-              variables,
-              operationName
-            )
+          context: contextFn,
+          routes: true,
+          additionalRouteProps: {
```
Comment (barelyhuman) [inline index.js:46]: should be `additionalRouteOptions`, I don't have push access to the repo anymore

Comment (simoneb) [reply index.js]: ah thanks, good catch 👌

**Pull request merged**